classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,408 @@
+            case self::ORDERBY_SEQUENCE:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'o.seq',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_PUBLISHED_ISSUES:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'o.seq',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'currentIssue',
+                        'direction' => self::ORDER_DIR_DESC
+                    ],
+                    [
+                        'orderBy' => 'i.date_published',
+                        'direction' => self::ORDER_DIR_DESC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_UNPUBLISHED_ISSUES:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'i.year',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.volume',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.number',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_SHELF:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => self::ORDER_CURRENT_ISSUE,
+                        'direction' => self::ORDER_DIR_DESC
+                    ],
+                    [
+                        'orderBy' => 'i.year',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.volume',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.number',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            default:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'i.date_published',
+                        'direction' => self::ORDER_DIR_DESC
+                    ]
+                ];
+                break;
+        }
+        return $this;
+    }
+
+    /**
+     * Set published filter
+     *
+     * @return $this
+     */
+    public function filterByPublished(bool $isPublished): self
+    {
+        $this->isPublished = $isPublished;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $volumes
+     *
+     * @return $this
+     */
+    public function filterByVolumes(array $volumes): self
```

Comment:
This should be `?array`, allowing calling code to reset the filter entirely (also below)